lib/max_subarray.py

```diff
@@ -1,12 +1,14 @@
-
 def max_sub_array(nums):
-    """ Returns the max subarray of the given list of numbers.
-        Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
-    """
+
     if nums == None:
         return 0
     if len(nums) == 0:
         return 0
-    pass
+
+    max_array = nums[0]
+    current_max = nums[0]
+
+    for i in range(1, len(nums)):
+        current_max = max(nums[i], current_max + nums[i])
```

Comment:
✨ This is a really nice way to represent this calculation, which captures the underlying invariant that makes Kadane's algorithm work. [This article](https://towardsdatascience.com/maximum-subarray-problem-and-kadanes-algorithm-4cb1ce91be72#8dc2) has a fairly good explanation of why this is considered a dynamic programming approach (on the face it might not "feel" like one).